scipy/optimize/_qap.py

```diff
@@ -0,0 +1,709 @@
+        msg = "'A' must be square"
+    elif B.shape[0] != B.shape[1]:
+        msg = "'B' must be square"
+    elif A.shape != B.shape:
+        msg = "Adjacency matrices must be of equal size"
+    elif (A < 0).any() or (B < 0).any():
+        msg = "Adjacency matrix contains negative entries"
+    elif partial_match.shape[0] > A.shape[0]:
+        msg = "There cannot be more seeds than there are nodes"
+    elif partial_match.shape[1] != 2:
+        msg = "`partial_match` must have two columns"
+    elif (partial_match < 0).any():
+        msg = "`partial_match` contains negative entries"
+    elif (partial_match >= len(A)).any():
+        msg = "`partial_match` entries must be less than number of nodes"
+    elif not len(set(partial_match[:, 0])) == len(partial_match[:, 0]) or not \
+            len(set(partial_match[:, 1])) == len(partial_match[:, 1]):
+        msg = "`partial_match` column entries must be unique"
+
+    if msg is not None:
+        raise ValueError(msg)
+
+    if not isinstance(maximize, bool):
+        msg = "'maximize' must be a boolean"
+
+    if msg is not None:
+        raise TypeError(msg)
+
+    return A, B, partial_match
+
+
+def _quadratic_assignment_faq(A, B,
+                              maximize=False, partial_match=None, rng=None,
+                              init_J="barycenter", init_weight=None, init_k=1,
+                              maxiter=30, shuffle_input=True, eps=0.05,
+                              **unknown_options
+                              ):
+    r"""
+    Solve the quadratic assignment problem (approximately).
+
+    This function solves the Quadratic Assignment Problem (QAP) and the
+    Graph Matching Problem (GMP) using the Fast Approximate QAP Algorithm
+    (FAQ) [1]_.
+
+    Quadratic assignment solves problems of the following form:
+
+    .. math::
+
+        \min_P & \ {\ \text{trace}(A^T P B P^T)}\\
+        \mbox{s.t. } & {P \ \epsilon \ \mathcal{P}}\\
+
+    where :math:`\mathcal{P}` is the set of all permutation matrices,
+    and :math:`A` and :math:`B` are adjacency matrices.
+
+    Graph matching tries to *maximize* the same objective function.
+    This algorithm can be thought of as finding the alignment of the
+    nodes of two graphs that minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements.
+
+    Note that the quadratic assignment problem is NP-hard, is not
+    known to be solvable in polynomial time, and is computationally
+    intractable. Therefore, the results given are approximations,
+    not guaranteed to be exact solutions.
+
+
+    Parameters
+    ----------
+    A : 2d-array, square, non-negative
+        The square adjacency matrix :math:`A` in the objective function above.
+
+    B : 2d-array, square, non-negative
+        The square adjacency matrix :math:`B` in the objective function above.
+
+    method :  str in {'faq', '2opt'}
```

Comment:
can be removed for this function, right?

Reply:
I'm not sure I understand. If you mean "Is it optional?", I'm adding a note about the default being 'faq'.

Reply:
I just meant this is the docstring for the FAQ function specifically, so this "method" parameter isn't used here. looked like a holdover from `quadratic_assignment` function

Reply:
When CI finishes, take a look at the "build artifacts" from CircleCI so you can see how the documentation renders. While this is written in `_quadratic_assignment_faq`, it is really the documentation for `quadratic_assignment` with `'method': 'faq'`.